packages/vc-verification/src/revocation-verification.ts

```diff
@@ -0,0 +1,107 @@
+import { addressOf } from '@ew-did-registry/did-ethr-resolver';
+import { verifyCredential } from 'didkit-wasm-node';
+import { AuthorityResolver, CredentialResolver } from '.';
+import { StatusPurpose, StatusListCredential } from './models';
+import { VCIssuerVerification } from './issuer-verification-vc';
+
+/**
+ * A class to validate if the revocation is valid or not
+ */
+export class RevocationVerification {
+  private _authorityResolver: AuthorityResolver;
+  private _credentialResolver: CredentialResolver;
+  private _issuerVerification: VCIssuerVerification;
+
+  /**
+   *
+   * @param revokerDefResolver
+   * @param credentialResolver
+   */
+  constructor(
+    authorityResolver: AuthorityResolver,
+    credentialResolver: CredentialResolver
+  ) {
+    this._authorityResolver = authorityResolver;
+    this._credentialResolver = credentialResolver;
+    this._issuerVerification = new VCIssuerVerification(
+      credentialResolver,
+      authorityResolver
+    );
+  }
+
+  /**
+   * Verify revocation for the provided credential or namespace and revoker
+   * @param credential
+   * @param namespace
+   * @param revokerDID
+   * @returns
+   */
+  async verifyRevocation(namespace: string, credential: StatusListCredential) {
```

Comment:
IMO, the function name is appropriate as `verifyRevokerAuthority` method performs the authorisation validation for the revoker (called from `verifyRevocation` along with `statusPurpose` check). Also, once we have the low-level `StatusListCredential` verification implemented in `ew-did-registry` then it would be used in the `verifyRevocation` method fulfilling the complete verification of revocation need.